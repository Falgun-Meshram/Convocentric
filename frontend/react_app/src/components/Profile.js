import React, { useState, useEffect } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { useNavigate } from "react-router-dom";
import '../css/Profile.css';
import { defaultProfilePictureImageDataUri } from "../constants";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCamera } from "@fortawesome/free-solid-svg-icons"
// import Dialog from "react-bootstrap-dialog";
import axiosInstance from './axiosInstance';
import { Spinner } from 'react-bootstrap';

export default function Profile() {

    // let CustomDialog = useRef(null);
    const [loading, setLoading] = useState(false);
    const navigate = useNavigate();
    const [editStatusMsg, setEditStatusMsg] = useState('');
    const [editDataError, setEditDataError] = useState('');
    const [validated, setValidated] = useState(false);
    const [formData, setFormData] = useState({
        id:'',
        password:'',
        first_name: '',
        last_name: '',
        email :'',
        profile_picture: '',
        username: '',
    })
    const [originalFormData, setOriginalFormData] = useState({
        id:'',
        password:'',
        first_name: '',
        last_name: '',
        email :'',
        profile_picture: '',
        username: '',
    })

    useEffect(() => {
        let currentFormData = {};
        if(localStorage.getItem('user')){
            let user = JSON.parse(localStorage.getItem('user'));
            currentFormData = {...user}
        }
        setFormData(currentFormData);
        setOriginalFormData(currentFormData);
    }, []);

    const redirectPage = (page) => {
        navigate(page)
    }

    const logout = () => {
        axiosInstance.get("logout/").then((response) => {
            if (response.data.ok) {
                localStorage.removeItem("isAuth");
                localStorage.removeItem("user");
                redirectPage('/');
            } else {
                console.log("Error");
            }
        }).catch((error) => {
            console.log(error);
        });
    }

    const shallowEqual = (object1, object2) => {
        const keys1 = Object.keys(object1);
        const keys2 = Object.keys(object2);
        if (keys1.length !== keys2.length) {
          return false;
        }
        for (let key of keys1) {
          if (object1[key] !== object2[key]) {
            return false;
          }
        }
        return true;
      }

    const handleSubmit = (e) => {

        e.preventDefault();
        if(shallowEqual(formData, originalFormData)){
            setEditDataError(true);
            setEditStatusMsg("You have not changed any values.");
            setTimeout(function () {
                setEditStatusMsg("");
            }, 5000);
        }else if (e.currentTarget.checkValidity() === false) {
            e.stopPropagation();
            e.preventDefault();
            setValidated(true);
        }else{
            const options = {
                method: 'PUT',
                url: 'manage_user/'+formData.id+'/',
                data: formData
            };
            axiosInstance.request(options).then(function (response) {
                setLoading(false);
                if(response.status === 204){
                    setFormData(formData);
                    setEditStatusMsg('User data saved successfully.');
                    setEditDataError(false);
                    setTimeout(() => {
                        setEditStatusMsg('');
                    }, 5000);
                    localStorage.setItem('user', JSON.stringify(formData))
                }else{
                    setEditDataError(true);
                    setEditStatusMsg(response.data.error);
                    setTimeout(() => {
                        setEditStatusMsg('');
                    }, 5000);
                }
            }).catch(function (error) {
                console.error(error);
            });
        }
    }

    const getBase64 = (file, cb) => {
        let reader = new FileReader();
        reader.readAsDataURL(file);
            reader.onload = function () {
            cb(reader.result)
        };
        reader.onerror = function (error) {
            console.log('Error: ', error);
        };
    }
    
    const profilePictureHandler = (e) => {
        let profile_picture_file = e.target.files[0];

        getBase64(profile_picture_file, (result) => {
            setFormData({...formData, profile_picture: result});
        });
    }

    // const removeProfilePicture = () => {
    //     setFormData({...formData, profile_picture: ''});
    // }
    
    // const removeProfilePictureRef = () => {
    //     CustomDialog.show({
    //         body:"Are you sure you want to remove your profile picture ?",
    //         actions: [
    //             Dialog.DefaultAction(
    //                 "Remove",
    //                 () => {removeProfilePicture()},
    //                 "btn-danger"
    //             ),
    //             Dialog.Action(
    //                 "Close",
    //                 () => {
    //                 if(CustomDialog){
    //                     CustomDialog.hide()
    //                 }
    //                 },
    //                 "btn-primary"
    //             ),
    //             ],
    //         });
    // };

    return (
        <Row style={{ padding: '0px', margin: '10% 0%' }}>
            <Col className="shadow-lg" style={{ backgroundColor: '#F6F6F6', border: '1px solid #ebebeb', borderRadius: '20px', padding: '10px 20px' }} sm={12} md={12} lg={{offset: 3, span: 6}} xl={{offset: 3, span: 6}}>
                 {/* <Row style={{ padding: '0px 0px', margin: '0px 0px 30px 0px' }}>
                        <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                            <p style={{ marginBottom: '0px', textAlign: 'left', color: "#0A194E", fontSize: '22px' }}>Profile Settings</p>
                            <div style={{ display: "none" }}>
                                <Dialog
                                ref={(el) => {
                                CustomDialog = el;
                                }}
                                />
                            </div>
                        </Col>
                    </Row> */}
                <Row> 
                    <Col sm={12} md={2} lg={2} xl={2}> 
                    <div style={{ position: 'relative' }}>
                        <img alt="profile_photo" src={formData && formData.profile_picture ? formData.profile_picture : defaultProfilePictureImageDataUri} className="circular_square"/>
                        <div className="profile_picture_upload">
                            <label htmlFor="profile_picture">
                                <FontAwesomeIcon color="gray" icon={faCamera} className="profile_picture_change"/>
                            </label>
                            <input onChange={profilePictureHandler} id="profile_picture" type="file" />
                            </div>
                        </div>
                    </Col>
                    <Col sm={12} md={2} lg={2} xl={2}>
                        <p style={{ textAlign: 'left', color: "#2A4CBF", fontSize: '20px', margin: '8px 0px 0px 10px' }}>@{formData.username}</p>
                        <p style={{ cursor:'pointer', textAlign: 'left', fontSize: '18px', margin: '2px 5px 0px 15px', fontWeight: 'bold', color: "#162B71" }} onClick={() => logout()}> Logout</p>
                    </Col>
                </Row>
                <Row style={{ padding: '0px', margin: '0px 0px 20px 0px' }}>
                    <Col sm={12} md={12} lg={12} xl={12} style={{ marginBottom: '10px' }}>
                        <p style={{ textAlign: 'left', paddingLeft: '10px', color: 'black', fontWeight: 'normal', fontSize: '18px' }}>Edit Details</p>
                    </Col>
                    <Col sm={12} md={12} lg={12} xl={12}>
                        <Form noValidate validated={validated} onSubmit={(e) => handleSubmit(e)}>
                            <Row style={{ margin: '0px', padding: '0px' }}>
                                <Col sm={4} md={4} lg={4} xl={4}>
                                    <Form.Group className="mb-3">
                                        <Form.Control required onChange={(e) => setFormData({ ...formData, first_name: e.target.value })} id="first_name" name="first_name" value={formData.first_name || ''} type="text" placeholder="First name" />
                                        <Form.Control.Feedback type="invalid">Please provide a valid firstname</Form.Control.Feedback>
                                    </Form.Group> 
                                </Col>
                                <Col sm={4} md={4} lg={4} xl={4}>
                                    <Form.Group className="mb-3">
                                        <Form.Control required onChange={(e) => setFormData({ ...formData, last_name: e.target.value })} id="last_name" name="last_name" value={formData.last_name || ''} type="text" placeholder="Last name" />
                                        <Form.Control.Feedback type="invalid">Please provide a valid lastname</Form.Control.Feedback>
                                    </Form.Group> 
                                </Col>
                            </Row> 
                            <Row style={{ margin: '0px', padding: '0px' }}>
                                <Col sm={8} md={8} lg={8} xl={8}>
                                    <Form.Group className="mb-3">
                                        <Form.Control required pattern='[\w\.]+@([\w]+\.)+[\w]{2,4}$' onChange={(e) => setFormData({ ...formData, email: e.target.value })} id="email" name="email" value={formData.email || ''} type="email" placeholder="Enter your email" />
                                        <Form.Control.Feedback type="invalid">Please provide a valid email id</Form.Control.Feedback>
                                    </Form.Group>
                                </Col> 
                            </Row>   
                            <Row style={{ margin: '0px', padding: '0px' }}>
                                <Col sm={8} md={8} lg={8} xl={8}>
                                    <Form.Group className="mb-3">
                                        <Form.Control disabled onChange={(e) => setFormData({ ...formData, username: e.target.value })} id="username" name="username" value={formData.username || ''} type="text" placeholder="Enter username" />
                                    </Form.Group>
                                </Col> 
                            </Row>   
                            {editStatusMsg?<Row style={{ margin: '0px', padding: '0px' }}>
                                <Col sm={8} md={8} lg={8} xl={8}>
                                    <p style={{ margin: '0px 0px 10px 0px', color: editDataError?'red':'green'}}>{editStatusMsg}</p>
                                </Col> 
                            </Row>:''}                 
                            {loading ?
                            <Row style={{ margin: '10px 0px 0px 0px', padding: '0px' }}>
                                <Col lg={{span: 2, offset: 3}} xl={{span: 2, offset: 3}}>
                                    <div className='text-center'>
                                        <Spinner animation="border" />
                                    </div>
                                </Col>
                            </Row>:
                            <Row>
                                <Col lg={4} xl={4}>
                                    <Button size="md" className="customButton" type="submit">
                                        Save
                                    </Button>
                                </Col>
                                <Col lg={4} xl={4}>
                                    <Button size="md" className="customButtonChangePassword" onClick={() => redirectPage('/home')} >
                                        Back to Home 
                                    </Button>
                                </Col>
                            </Row>}

                            {/* // <Col lg={4} xl={4}>
                            //     <Button size="md" className="customButtonChangePassword" onClick={() => redirectPage('/change_password')} >
                            //         Change Password
                            //     </Button>
                            // </Col>  */}
                        </Form>
                    </Col>
                </Row>
            </Col> 
        </Row>
    )
}