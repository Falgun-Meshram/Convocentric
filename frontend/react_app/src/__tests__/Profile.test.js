import { render, fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import React from 'react';
import { setupServer } from 'msw/node';
import { response, rest } from 'msw'
import { BrowserRouter } from 'react-router-dom';

import Profile from '../components/Profile';
import Signin from '../components/Signin';

const server = setupServer(
    rest.put("http://127.0.0.1:8000/api/manage_user/1", (req, res, ctx) => {
        return res(
            ctx.status(204),
            ctx.json({
                ok: true,
                status: 204

            }))
    })
)
const user = '{ "id": 1, "password": "pbkdf2_sha256$320000$GQuoWffBrr1vLvXLgbTCd1$9cfnIcqpnq6B+02cp5AhlwvLaaCKLenulqK9/e4JxQ8=", "is_superuser": false, "is_staff": false, "is_active": true, "date_joined": "2022-03-17T16:12:04.037632Z", "first_name": "test", "last_name": "test", "email": "test1@test.com", "created_on": "2022-03-17T16:10:41.719550Z", "updated_on": "2022-03-17T16:10:41.719723Z", "profile_picture": "", "last_passwords": null, "locked": null, "last_login": "2022-03-31T18:44:50.351243Z", "username": "test1", "groups": [], "user_permissions": [] }'

beforeAll(() => {
    localStorage.setItem('user', user)
    server.listen()
})
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

describe("<Profile>", () => {
    it("Page should render", async () => {
        const { container, debug } = render(
            <BrowserRouter>
                <Profile />
            </BrowserRouter>
        );
        const userName = await screen.findAllByText("@test1")
        console.log(userName)
        expect(userName[0]).toBeInTheDocument();
        expect(container).toMatchSnapshot();

    });
    it("Profile firstname, last name and email validation", async () => {
        const { container, debug } = render(
            <BrowserRouter>
                <Profile />
            </BrowserRouter>
        );

        const lastname = screen.getByPlaceholderText("Last name");
        const firstname = screen.getByPlaceholderText("First name");
        const email = screen.getByPlaceholderText("Enter your email");
        const input = screen.getByTestId("upload-input")
        const file = new File(['hello'], 'hello.png', { type: 'image/png' })

        userEvent.type(lastname, "81")
        userEvent.type(firstname, "test")
        await userEvent.upload(input, file)
        userEvent.click(screen.getByText("Save"))
        const c = await screen.findByText("User data saved successfully.");
        expect(c).toBeInTheDocument();

    }
    )
    it("Check the logged in user", async () => {
        const { container, debug } = render(
            <BrowserRouter>
                <Profile />
            </BrowserRouter>
        );
        const firstname = await screen.findByDisplayValue("test1");
        expect(firstname).toBeInTheDocument();
    }
    )
    it("check the error  message", async () => {
        const { container, debug } = render(
            <BrowserRouter>
                <Profile />
            </BrowserRouter>
        );
        const save_button = screen.getByText("Save");
        userEvent.click(screen.getByText("Save"))
        const c = await screen.findByText("You have not changed any values.")
        expect(c).toBeInTheDocument();
    }
    )


    it("Test logout function", async () => {
        server.use(
            rest.post('http://127.0.0.1:8000/api/logout', (req, res, ctx) => {
                return res({ 'ok': true })
            }),
        )
        const { container, debug } = render(
            <BrowserRouter>
                <Profile />
                <Signin />
            </BrowserRouter>
        );

        const log_out = screen.getByText("Logout");
        userEvent.click(log_out)
        const c = screen.getByText("Convocentric");
        expect(c).toBeInTheDocument();
    }
    );
    it("User update network fail", async () => {
        server.use(
            rest.put("http://127.0.0.1:8000/api/manage_user/1", (req, res, ctx) => {
                return res(
                    ctx.status(202),
                    ctx.json({
                        error: "Request Accepted",
                    }))
            })
        )
        const { container, debug } = render(
            <BrowserRouter>
                <Profile />
            </BrowserRouter>
        );
        const lastname = screen.getByPlaceholderText("Last name");

        userEvent.type(lastname, "81")
        userEvent.click(screen.getByText("Save"))
        const c = await screen.findByText("Request Accepted")
        expect(c).toBeInTheDocument();
    }
    );

})

