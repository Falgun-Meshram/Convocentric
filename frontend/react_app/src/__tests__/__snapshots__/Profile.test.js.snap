// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> Page should render 1`] = `
<div>
  <div
    class="row"
    style="padding: 0px; margin: 10% 0%;"
  >
    <div
      class="shadow-lg col-xl-6 col-lg-6 col-md-12 col-sm-12 offset-xl-3 offset-lg-3"
      style="background-color: rgb(246, 246, 246); border: 1px solid #ebebeb; border-radius: 20px; padding: 10px 20px;"
    >
      <div
        class="row"
      >
        <div
          class="col-xl-2 col-lg-2 col-md-2 col-sm-12"
        >
          <div
            style="position: relative;"
          >
            <img
              alt="profile_photo"
              class="circular_square"
              src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEBLAEsAAD/4QBWRXhpZgAATU0AKgAAAAgABAEaAAUAAAABAAAAPgEbAAUAAAABAAAARgEoAAMAAAABAAIAAAITAAMAAAABAAEAAAAAAAAAAAEsAAAAAQAAASwAAAAB/+0ALFBob3Rvc2hvcCAzLjAAOEJJTQQEAAAAAAAPHAFaAAMbJUccAQAAAgAEAP/hDIFodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvADw/eHBhY2tldCBiZWdpbj0n77u/JyBpZD0nVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkJz8+Cjx4OnhtcG1ldGEgeG1sbnM6eD0nYWRvYmU6bnM6bWV0YS8nIHg6eG1wdGs9J0ltYWdlOjpFeGlmVG9vbCAxMC4xMCc+CjxyZGY6UkRGIHhtbG5zOnJkZj0naHR0cDovL3d3dy53My5vcmcvMTk5OS8wMi8yMi1yZGYtc3ludGF4LW5zIyc+CgogPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9JycKICB4bWxuczp0aWZmPSdodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyc+CiAgPHRpZmY6UmVzb2x1dGlvblVuaXQ+MjwvdGlmZjpSZXNvbHV0aW9uVW5pdD4KICA8dGlmZjpYUmVzb2x1dGlvbj4zMDAvMTwvdGlmZjpYUmVzb2x1dGlvbj4KICA8dGlmZjpZUmVzb2x1dGlvbj4zMDAvMTwvdGlmZjpZUmVzb2x1dGlvbj4KIDwvcmRmOkRlc2NyaXB0aW9uPgoKIDxyZGY6RGVzY3JpcHRpb24gcmRmOmFib3V0PScnCiAgeG1sbnM6eG1wTU09J2h0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8nPgogIDx4bXBNTTpEb2N1bWVudElEPmFkb2JlOmRvY2lkOnN0b2NrOjMyOTBjNTNjLWI1NmEtNGU0OC05YzIwLTRmNzYzZDU4N2Q3MjwveG1wTU06RG9jdW1lbnRJRD4KICA8eG1wTU06SW5zdGFuY2VJRD54bXAuaWlkOmNlMDA3MDk5LWVkYjYtNGRkYi1hNTFmLTY2NzMzZDM0MWQwZDwveG1wTU06SW5zdGFuY2VJRD4KIDwvcmRmOkRlc2NyaXB0aW9uPgo8L3JkZjpSREY+CjwveDp4bXBtZXRhPgogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAo8P3hwYWNrZXQgZW5kPSd3Jz8+/9sAQwADAgIDAgIDAwMDBAMDBAUIBQUEBAUKBwcGCAwKDAwLCgsLDQ4SEA0OEQ4LCxAWEBETFBUVFQwPFxgWFBgSFBUU/8AACwgA8ADwAQERAP/EABwAAQACAwEBAQAAAAAAAAAAAAAHCAEFBgIEA//EADkQAAEDAgIFCAkEAwEAAAAAAAABAgMEBQYRByExQVESFDZxdIGRshMiJENicqGxwRYjYdEyU4Iz/9oACAEBAAA/ALNgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAy1jn/4tV3UmYcxzP8mq3rTIwAAAAAAAAAAAAeoonzyNjjY6SRy5Na1M1Vf4QkfC+hauuLWT3aXmEK6/QtTOVU/nc0km06N8PWdrfR26OaRPeVH7jl8dR0EVHTwN5McEcacGMRBLR087eTLBHIi7nsRTn7to3w9eGu9Jbo4ZF95T/tuTw1EbYo0LV1ua+e0y8/hTWsLkylTq3OI4lifBI6ORjo5Grk5rkyVF/lDyAAAAAAAAAAfvQUFRdKyKlpYnTVEruSxjd6lgMB6OqTCVO2eZG1NzcnrzKmaM/hvDr3nYgAA47HmjqkxbTunhRtNc2p6kyJkj/wCHcevcV/r6CotdZLS1UToaiJ3Jex25T8AAAAAAAAAATlohwW20Wxt2qo/bapucaOTXHHu712+BIwAAAI50vYLbd7Y67UsfttK3ORGprkj396bfEg0AAAAAAAAA3mCrF+o8T0NE5M4nP5cvyN1r/XeWaY1GNRrURrUTJETch6AAAB5e1HtVrkRzVTJUXehWXGti/TmJ66iamUTX8uL5Ha0/ruNGAAAAAAAACUNBFAkt2uVYqZrFE2Nq8Fcua+UmgAAAAEL6d6BIrtbaxEyWWJ0bl4q1c08xF4AAAAAAAAJi0C5czvHH0kf2UlYAAAAEU6esuZ2fj6ST7IQ6AAAAAAAACUtA9ckdyulIq5LJE2RqceSuS+YmYAAAAEM6eK5JLla6RFzWOJ0jk4cpck8pFoAAAAAAAAOgwHfUw7iqhq3u5MKu9HL8rtS+G3uLLoqORFRc0XeZAAABhVRqKqrkibytGPL6mIsVV1Wx3KhR3o4vlbqTx295z4AAAAAAAABPWibGTb9Z22+ok9vo2o3WuuSPcvdsU74AAAHA6WcZNsNndb6eT2+sardS644969+xCBQAAAAAAAAAfXarrU2S4Q1tHIsVREubXJ9l4opYXBOPKLGFGiNVsFexP3aZV19beKHUAAA5fG2PKLB9GqOVs9e9P2qZF19buCFertdam93CatrJFlqJVzc5fsnBEPkAAAAAAAAAAP1pauahqGT08r4ZmLm2Ri5Kiko4X03SQMZBe4FmRNXOoE9b/pu/uJItOM7Je2otJcYHuX3bncl/gus3KORyZoqKnFArkamaqiJxU012xnZLI1Vq7jAxye7a7lP8E1kb4o03STsfBZIFhRdXOp09b/lu7vIuqquauqHz1Er5pnrm6R65qqn5AAAAAAAAAAAAH0Q3Orp0yiq5404MkVPyJrnV1CZS1c8icHyqv5PnAAAAAAAAAAAMtar3I1qK5y6kREzVTq7JovxBe0a9tHzSFfeVS8j6bfodrbNBEDUR1wub5F3sp2I1PFc/sdLRaI8M0mXKon1Kpvmlcv0TJDbQYIsFNl6Oz0aZcYkX7n1tw7amJ6tso06oG/0HYdtT09a2Ua9cDf6PknwRYKn/ANLPRr1RIn2NTW6I8M1efJon0yrvhlcn0XNDmrnoIgciut9zfGu5lQxHJ4pl9jir3ovxBZEc91HzuFPeUq8v6bfoco5qscrXIrXJqVFTJUMAAAAAAAAHYYM0Z3HFfJnf7Fb/APe9Nb/lTf17CZ8N4Fs+F2N5pStfPlrqJfWkXv3dx0IAAAAOexJgWz4oY7ndK1k+Wqoi9WRO/f3kMYz0Z3HCnKnZ7bb/APexNbPmTd17DjwAAAAAACTdGGjNLskd2usfseecFO73vxL8P3JpYxsbGtY1GtamSNRMkRD0AAAAADy9jZGOa9qOa5MlaqZoqELaTtGaWlJLtao/Y8856dvuviT4fsRkAAAAAAdPo7wp+rMQxwyIvM4U9LOvFu5vev5LHxxthjbGxqMY1Ea1rUyRE4HoAAAAAAHmSNs0bo3tR7HIrXNcmaKnArhpEwp+k8QyQxovM5k9LAvBu9vcv4OYAAAAABPOhezNoMKLWK3KWtkV+fwpqT8r3nfgAAAAAAA4DTRZm1+FErEbnLRSI/P4V1L+F7iBgAAAAAWdwRClPhCzsbqTmrF8Uz/JuwAAAAAAAaTG0KVGELwx2tOavXwTP8FYgAAAADC7C0WEei1o7LF5UNuAAAAAAADUYu6LXfssvlUq6mwyAAAADC7C0WEei1o7LF5UNuAAAAAAADUYu6LXfssvlUq6mwyAAAADC7C0WEei1o7LF5UNuAAAAAAADUYu6LXfssvlUq6mwyAAAADC7C0WEei1o7LF5UNuAAAAAAADUYu6LXfssvlUq6mwyAAAADC7C0WEei1o7LF5UNuAAAAAAADUYu6LXfssvlUq6mwyAAAADC7C0WEei1o7LF5UNuAAAAAAADUYu6LXfssvlUq6mwyAAf/Z"
            />
            <div
              class="profile_picture_upload"
            >
              <label
                for="profile_picture"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-camera profile_picture_change"
                  color="gray"
                  data-icon="camera"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M194.6 32H317.4C338.1 32 356.4 45.22 362.9 64.82L373.3 96H448C483.3 96 512 124.7 512 160V416C512 451.3 483.3 480 448 480H64C28.65 480 0 451.3 0 416V160C0 124.7 28.65 96 64 96H138.7L149.1 64.82C155.6 45.22 173.9 32 194.6 32H194.6zM256 384C309 384 352 341 352 288C352 234.1 309 192 256 192C202.1 192 160 234.1 160 288C160 341 202.1 384 256 384z"
                    fill="currentColor"
                  />
                </svg>
              </label>
              <input
                data-testid="upload-input"
                id="profile_picture"
                type="file"
              />
            </div>
          </div>
        </div>
        <div
          class="col-xl-2 col-lg-2 col-md-2 col-sm-12"
        >
          <p
            style="text-align: left; color: rgb(42, 76, 191); font-size: 20px; margin: 8px 0px 0px 10px;"
          >
            @
            test1
          </p>
          <p
            style="cursor: pointer; text-align: left; font-size: 18px; margin: 2px 5px 0px 15px; font-weight: bold; color: rgb(22, 43, 113);"
          >
             Logout
          </p>
        </div>
      </div>
      <div
        class="row"
        style="padding: 0px; margin: 0px 0px 20px 0px;"
      >
        <div
          class="col-xl-12 col-lg-12 col-md-12 col-sm-12"
          style="margin-bottom: 10px;"
        >
          <p
            style="text-align: left; padding-left: 10px; color: black; font-weight: normal; font-size: 18px;"
          >
            Edit Details
          </p>
        </div>
        <div
          class="col-xl-12 col-lg-12 col-md-12 col-sm-12"
        >
          <form
            class=""
            novalidate=""
          >
            <div
              class="row"
              style="margin: 0px; padding: 0px;"
            >
              <div
                class="col-xl-4 col-lg-4 col-md-4 col-sm-4"
              >
                <div
                  class="mb-3"
                >
                  <input
                    class="form-control"
                    id="first_name"
                    name="first_name"
                    placeholder="First name"
                    required=""
                    type="text"
                    value="test"
                  />
                  <div
                    class="invalid-feedback"
                  >
                    Please provide a valid firstname
                  </div>
                </div>
              </div>
              <div
                class="col-xl-4 col-lg-4 col-md-4 col-sm-4"
              >
                <div
                  class="mb-3"
                >
                  <input
                    class="form-control"
                    id="last_name"
                    name="last_name"
                    placeholder="Last name"
                    required=""
                    type="text"
                    value="test"
                  />
                  <div
                    class="invalid-feedback"
                  >
                    Please provide a valid lastname
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
              style="margin: 0px; padding: 0px;"
            >
              <div
                class="col-xl-8 col-lg-8 col-md-8 col-sm-8"
              >
                <div
                  class="mb-3"
                >
                  <input
                    class="form-control"
                    id="email"
                    name="email"
                    pattern="[\\\\w\\\\.]+@([\\\\w]+\\\\.)+[\\\\w]{2,4}$"
                    placeholder="Enter your email"
                    required=""
                    type="email"
                    value="test1@test.com"
                  />
                  <div
                    class="invalid-feedback"
                  >
                    Please provide a valid email id
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
              style="margin: 0px; padding: 0px;"
            >
              <div
                class="col-xl-8 col-lg-8 col-md-8 col-sm-8"
              >
                <div
                  class="mb-3"
                >
                  <input
                    class="form-control"
                    disabled=""
                    id="username"
                    name="username"
                    placeholder="Enter username"
                    type="text"
                    value="test1"
                  />
                </div>
              </div>
            </div>
            
            <div
              class="row"
            >
              <div
                class="col-xl-4 col-lg-4"
              >
                <button
                  class="customButton btn btn-primary btn-md"
                  type="submit"
                >
                  Save
                </button>
              </div>
              <div
                class="col-xl-4 col-lg-4"
              >
                <button
                  class="customButtonChangePassword btn btn-primary btn-md"
                  type="button"
                >
                  Back to Home
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
